mypy/semanal.py

```diff
@@ -871,6 +868,11 @@ def add_function_to_symbol_table(self, func: Union[FuncDef, OverloadedFuncDef])
     def analyze_function_body(self, defn: FuncItem) -> None:
         is_method = self.is_class_scope()
         with self.tvar_scope_frame(self.tvar_scope.method_frame()):
+            # Analyze default arguments
+            for arg in defn.arguments:
+                if arg.initializer:
+                    arg.initializer.accept(self)
```

Comment:
I assume you move this here because this method is already called on lambdas. I however see two problems with this:
* The comment on line 560 above becomes outdated (can't comment so far above).
* This can potentially cause problem with scopes if make some refactorings in future, so I would add a test like this (if we don't have one already):
```python
def f(x=i):  # Should give a 'Name not defined' error
    i = 42
```